### PR TITLE
chore: resolve warnings from yaml and numpy, fix supposed fstrings, black formatting

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

Fix some warnings from `yaml` and `numpy`.
Fix a few fstrings missing the prefix `f`.
Fix some typing mismatches, solving several `mypy` errors.
Fix a little bit of code formatting and typos.

## Test Plan

To see the reduction in `numpy` warnings, in the directory `determined/harness`, running the command
`pytest tests/experiment/pytorch/test_reducer.py tests/experiment/tensorflow/test_estimator_trial.py`
would result in 4140 warnings. After the proposed changes, this reduces to 29 warnings. There are certainly repeated warnings.

The output of `mypy` as used in, say, `make -C harness check` is now as follows:
```
determined/pytorch/_reducer.py:36: error: Returning Any from function declared to return "floating[_64Bit]"
determined/pytorch/_reducer.py:38: error: Returning Any from function declared to return "floating[_64Bit]"
determined/pytorch/_reducer.py:40: error: Returning Any from function declared to return "floating[_64Bit]"
determined/pytorch/_reducer.py:42: error: Returning Any from function declared to return "floating[_64Bit]"
determined/ipc.py:68: error: Call to untyped function "Context" in typed context
determined/ipc.py:70: error: Call to untyped function "socket" in typed context
determined/ipc.py:71: error: Call to untyped function "socket" in typed context
determined/ipc.py:175: error: Call to untyped function "Context" in typed context
determined/ipc.py:177: error: Call to untyped function "socket" in typed context
determined/ipc.py:182: error: Call to untyped function "socket" in typed context
Found 10 errors in 2 files (checked 277 source files)
```
The solutions to these remaining errors are not clear (and writing `# type: ignore` everywhere doesn't seem appropriate in these places), so they will be left for now, unless someone has a suggestion!

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by drh-determined-ai** [comment]
The message for commit [d122643](https://github.com/determined-ai/determined/pull/4372/commits/d122643cb041cf311c91c5d38b0b0b71ca038ac7) should refer to `mypy`, not `lint`. The errors occur in the automated command `make -C harness check`.
